CMake/Findfmt.cmake

```
@@ -0,0 +1,32 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+find_path(LIBFMT_INCLUDE_DIR fmt/core.h)
+mark_as_advanced(LIBFMT_INCLUDE_DIR)
+
+find_library(LIBFMT_LIBRARY NAMES fmt fmtd)
+mark_as_advanced(LIBFMT_LIBRARY)
+
+include(FindPackageHandleStandardArgs)
+FIND_PACKAGE_HANDLE_STANDARD_ARGS(
+    LIBFMT
+    DEFAULT_MSG
+    LIBFMT_LIBRARY LIBFMT_INCLUDE_DIR)
+
+if(LIBFMT_FOUND)
+    set(LIBFMT_LIBRARIES ${LIBFMT_LIBRARY})
+    set(LIBFMT_INCLUDE_DIRS ${LIBFMT_INCLUDE_DIR})
+    message(STATUS "Found {fmt}: ${LIBFMT_LIBRARY}")
```

I think it would be nicer to have Findfmt.cmake also define a newer-style `fmt::fmt` target.  This way the calling code can depend on this target regardless of whether it finds fmt through config-fmt.cmake or through Findfmt.cmake.

```suggestion
    set(LIBFMT_LIBRARIES ${LIBFMT_LIBRARY})
    set(LIBFMT_INCLUDE_DIRS ${LIBFMT_INCLUDE_DIR})
    add_library(fmt::fmt UNKNOWN IMPORTED)
    set_target_properties(fmt::fmt PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${LIBFMT_INCLUDE_DIR}")
    set_target_properties(fmt::fmt PROPERTIES IMPORTED_LINK_INTERFACE_LANGUAGES "C" IMPORTED_LOCATION "${LIBFMT_LIBRARY}")
    message(STATUS "Found {fmt}: ${LIBFMT_LIBRARY}")
```

---

Note that our FindGflags.cmake and FindGlog.cmake modules both work this way, although folly-deps.cmake isn't currently taking advantage of the new-style targets emitted by those modules.  We probably should update folly-deps.cmake to use the new-style targets for these libraries.  FindGflags.cmake internally takes care of checking for an installed CMake config for gflags, so that the caller can just use `find_package(Gflags MODULE)` and it does the right search internally.